pkgs/development/compilers/tinygo/default.nix

```diff
@@ -1,47 +1,70 @@
-{ lib, buildGoModule, fetchFromGitHub, llvm, clang-unwrapped, lld, avrgcc
-, avrdude, openocd, gcc-arm-embedded, makeWrapper, fetchurl }:
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, llvm
+, clang-unwrapped
+, lld
+, libffi
+, libclang
+, avrgcc
+, avrdude
+, openocd
+, gcc-arm-embedded
+, makeWrapper
+, fetchurl
+, substituteAll
+}:
 
-let main = ./main.go;
-    gomod = ./go.mod;
+let
+  main = ./main.go;
+  gomod = ./go.mod;
+  # Upstream PR: https://github.com/tinygo-org/tinygo/pull/2751.
+  go117patch = ./go117.patch;
+  clang-envpatch = substituteAll {
+    src = ./clang-env.patch;
+    libclang_inc = "${libclang.lib}/lib/clang/${lib.getVersion libclang}/include";
```

Comment:
A less ugly way of adding libclang as a `-isystem` path without patching the source is welcome.